personalize.api.php

```diff
@@ -56,15 +56,14 @@ function hook_personalize_option_set_type() {
 }
 
 /**
- * An agent is about to be created or updated.
- *
- * This hook is invoked before the agent is saved to the database so
- * modules can alter the information that is saved.
+ * Allows the alteration of an agent before it is saved.
  *
  * @param stdClass $agent
  *   A stdClass object representing the agent being saved.
+ * @param array $prior_values
+ *   The unalterable prior data for the agent before this save.
  */
-function hook_personalize_agent_presave($agent) {
+function hook_personalize_agent_alter(&$agent, $prior_values) {
```

Comment:
The information that was in the presave hook didn't matter anyway cause it was being set as defaults later on.  But this will matter.  I'll put it back to presave and actually invoke it then.  Of course an object is by reference - that fixes it all :-)